Core/Core/Assignments/AssignmentList/ViewModel/AssignmentCellViewModel.swift

```diff
@@ -30,30 +30,52 @@ public class AssignmentCellViewModel: ObservableObject {
     }
 
     public var route: URL? { assignment.htmlURL }
+    var icon: UIImage { assignment.icon ?? .assignmentLine }
+    var name: String { assignment.name }
+    var submissionStatus: String { assignment.submission?.stateDisplayProperties.text ?? SubmissionStatus.notSubmitted.text }
+    var submissionIcon: UIImage { assignment.submission?.stateDisplayProperties.icon ?? SubmissionStatus.notSubmitted.icon }
+    var submissionColor: Color { .init(assignment.submission?.stateDisplayProperties.color ?? SubmissionStatus.notSubmitted.color) }
+    let defaultTextColor: Color = .textDark
+    let brandColor: Color = .init(Brand.shared.primary)
+    var hasPointsPossible: Bool { scoreLabel != nil }
+    var pointsPossibleText: String { assignment.pointsPossibleCompleteText }
 
-    public var icon: UIImage {
-        assignment.icon ?? .assignmentLine
+    // Teacher
+    var isTeacher: Bool { env.app == .teacher }
+    var needsGrading: Bool { assignment.needsGradingCount > 0 }
+    var needsGradingCount: Int { assignment.needsGradingCount }
+
+    private var formattedDueAt: String? {
+        guard let dueAt = assignment.dueAt else { return nil }
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "LLL d, y h:mm a"
+        return dateFormatter.string(from: dueAt)
     }
 
-    public var name: String {
-        assignment.name
+    var scoreLabel: String? {
+        guard let pointsPossible = assignment.pointsPossible else { return nil }
```

Comment:
Just to doublecheck: will this cover the case when `hideQuantitativeData` is `true`?
Should that be added here?
summoning @vargaat for advice

Reply:
I just tested and this still shows the score when hide quantitative data is enabled so we have to add/copy/resuse the logic that handles it.